### PR TITLE
feat: add lib entrypoint exemple

### LINKED ISSUE #10: Prototype of Library EntryPoint
### Describe the feature

I have to develop entrypoint of the Libraries and get her type

### Example Feature Purpose

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AxelF44**
I added the examples of entry points that will have to be implemented for each library. It will therefore be necessary in the code first called the entrypoint which will allow to get the type of the libraries in order to know which entrypoint called.

### CODE REVIEW

File: types/LibraryType.hpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+** LibraryType
+*/
+
+#pragma once
+
+namespace shared::types
+{
+  enum LibraryType{
+    Game,
+    Renderer
```
Comment (flavien-chenu) [inline types/LibraryType.hpp]:
```suggestion
    GAME,
    RENDERER
```

File: exportRenderer.exemple [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+extern "C" LibraryType arcadeLibGetType()
+{
+    ...
+}
```
Comment (flavien-chenu) [inline exportRenderer.exemple]:
```suggestion
extern "C" shared::types::LibraryType arcadeLibGetType()
{
    return shared::types::LibraryType.RENDERER;
}
```

File: exportGame.exemple [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+extern "C" LibraryType arcadeLibGetType()
+{
+    ...
+}
```
Comment (flavien-chenu) [inline exportGame.exemple]:
```suggestion
```suggestion
extern "C" shared::types::LibraryType arcadeLibGetType(void)
{
    return shared::types::LibraryType.GAME;
}
```
```

File: exportGame.exemple [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+extern "C" std::unique_ptr<IGame> arcadeLibGetGame()
+{
+    ...
+}
```
Comment (flavien-chenu) [inline exportGame.exemple]:
```suggestion
extern "C" std::unique_ptr<shared::games::IGame> arcadeLibGetGame(void)
{
    return std::make_unique<GameClass>(...);
}
```
ℹ️ **Here `GameClass` will be replaced by target game class**

File: exportRenderer.exemple [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+extern "C" std::unique_ptr<IRenderer> arcadeLibGetRenderer()
+{
+    ...
+}
```
Comment (flavien-chenu) [inline exportRenderer.exemple]:
```suggestion
extern "C" std::unique_ptr<shared::graphic::IRenderer> arcadeLibGetRenderer(void)
{
    return std::make_unique<RendererClass>(...);
}
```
ℹ️ **Here `RendererClas` will be replaced by target renderer class**

File: exportGame.exemple
Comment (flavien-chenu) [inline exportGame.exemple:1]: Put your file in `games/export.cpp.example` **not at root with name `exportGame.exemple`**

File: exportRenderer.exemple
Comment (flavien-chenu) [inline exportRenderer.exemple:1]: Put your file in `graphic/export.cpp.example` **not at root with name `exportRenderer.exemple`**

**Review by flavien-chenu — CHANGES_REQUESTED** [review]

**Review by Yann-Masson — DISMISSED** [review]

**Review by Yann-Masson — DISMISSED** [review]

**Review by BxptisteM — DISMISSED** [review]

**Review by TekMath — CHANGES_REQUESTED** [review]
If you add shareptr for Game and Display, please update UniqueGame type to SharedGame type

**Review by sdragos1 — DISMISSED** [review]

**Review by Yann-Masson — DISMISSED** [review]

**Review by flavien-chenu — DISMISSED** [review]

**Review by Yann-Masson — DISMISSED** [review]

**Review by Yann-Masson — APPROVED** [review]

**Review by TekMath — APPROVED** [review]
Let's make it!

**Review by flavien-chenu — APPROVED** [review]

**Pull request merged**